src/EditorFeatures/Core/Implementation/Classification/SyntacticClassificationTaggerProvider.TagComputer.cs

```diff
@@ -176,422 +214,317 @@ public void DisconnectFromWorkspace()
-                    // we have the root of the tree ready to go.
-                    currentCachedData = service == null
-                        ? null
-                        : await service.GetDataToCacheAsync(currentDocument, cancellationToken).ConfigureAwait(false);
-
-                    // Query the service to determine waht span of the document actually changed and should be
-                    // reclassified in the host editor.
-                    changedSpan = await GetChangedSpanAsync(currentDocument, currentSnapshot, cancellationToken).ConfigureAwait(false);
-                }
-
-                lock (_gate)
-                {
-                    _lastProcessedSnapshot = currentSnapshot;
-                    _lastProcessedDocument = currentDocument;
-                    _lastProcessedCachedData = currentCachedData;
-                }
-
-                _notificationService.RegisterNotification(() =>
-                    {
-                        _workQueue.AssertIsForeground();
-                        ReportChangedSpan(changedSpan);
-                    },
-                    _listener.BeginAsyncOperation("EnqueueProcessSnapshotWorkerAsync.ReportChangedSpan"),
-                    _reportChangeCancellationSource.Token);
+                EnqueueWorkIfThisDocument(args.NewActiveContextDocumentId);
             }
 
-            private async Task<SnapshotSpan> GetChangedSpanAsync(
-                Document currentDocument, ITextSnapshot currentSnapshot, CancellationToken cancellationToken)
+            private void OnDocumentOpened(object? sender, DocumentEventArgs args)
             {
-                // We don't need to grab _lastProcessedDocument in a lock.  We don't care which version of the previous
-                // doc we grab, just that we grab some prior version.  This is only used to narrow down the changed range we 
-                // specify, so it's ok if it's slightly larger because we read in a change from a couple of edits ago.
-                var previousDocument = _lastProcessedDocument;
-                if (previousDocument != null)
-                {
-                    var service = TryGetClassificationService(currentSnapshot);
-                    if (service != null)
-                    {
-                        var changeRange = await service.ComputeSyntacticChangeRangeAsync(
-                            previousDocument, currentDocument, _diffTimeout, cancellationToken).ConfigureAwait(false);
-                        if (changeRange != null)
-                            return currentSnapshot.GetSpan(changeRange.Value.Span.Start, changeRange.Value.NewLength);
-                    }
-                }
-
-                // Couldn't compute a narrower range.  Just the mark the entire file as changed.
-                return currentSnapshot.GetFullSpan();
+                EnqueueWorkIfThisDocument(args.Document.Id);
             }
```

Comment:
not sure if this is needed at all.  @jasonmalinowski would we ever have to care bout doc-open?  definitionally we're a tagger.  so we only could ever be attached to an open doc...